ui/raidboss/timeline.ts

```diff
@@ -342,16 +352,45 @@ export class Timeline {
         durationEvents.push(event);
         continue;
       }
-      this.ui?.OnRemoveTimer(event, false, true);
+      this.ui?.OnRemoveTimer(event, true);
     }
+    // Do not clear keep alive events here, as this is part of a sync jump
+    // and keep alive timing is independent of timeline time.
 
     this.activeEvents = durationEvents;
   }
 
   private _RemoveExpiredTimers(fightNow: number): void {
     let activeEvent = this.activeEvents[0];
     while (this.activeEvents.length && activeEvent && activeEvent.time <= fightNow) {
-      this.ui?.OnRemoveTimer(activeEvent, true);
+      const event = activeEvent;
+      if (this.options.KeepExpiredTimerBarsForSeconds > 0) {
+        this.keepAliveEvents.push({
+          event: event,
+          timeout: window.setTimeout(
+            () => {
+              // Find and remove the first keepalive event with this id.
+              let found = false;
+              this.keepAliveEvents = this.keepAliveEvents.filter((x) => {
+                if (found)
+                  return true;
+                if (x.event.id === event.id) {
+                  found = true;
+                  return false;
+                }
```

Comment:
Does this need to `return true;` in the fallthrough where it is not found and the ids are not equal?

Reply:
Oh yeah, thanks!